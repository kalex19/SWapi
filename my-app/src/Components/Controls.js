import React, {Component} from 'react';
import Styles from '.Controls.scss';

export default Controls from Component {

//const ...

//filter functionality
//const Hello = ({name}) => <div>Hello {name}</div>

render () {

return(
<div>
<input>People</input>
<input>Planets</input>
<input>Vehicles</input>
</div>

)};

Controls.proptypes = {
  
//functionality
}


}


