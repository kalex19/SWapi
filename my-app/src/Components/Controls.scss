@import '../variables.scss';

.controls {
  align-items: center;
  margin-left: 20%;
}

.controls input {
  margin-left: 4%;
  background-color: gold;
  color: #000;
  border: 2px white solid;
  height: 50px;
  width: 200px;
  margin-top: 2%;
  font-size: 2em;
}