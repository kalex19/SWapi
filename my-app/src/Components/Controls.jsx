import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Controls.scss';

const Controls = props => {
	return (
		<div className="controls">
			<input className="fas fa-jedi" name="people" value="People" type="button" onClick={props.handleClick} />
			<input className="fas fa-globe" name="planets" value="Planets" type="button" onClick={props.handleClick} />
			<input
				className="fas fa-space-shuttle"
				name="vehicles"
				value="Vehicles"
				type="button"
				onClick={props.handleClick}
			/>
		</div>
	);
};

Controls.propTypes = {
	handlClick: PropTypes.func
};

export default Controls;
