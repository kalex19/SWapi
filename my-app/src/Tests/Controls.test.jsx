import React from 'react';
import { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import Controls from '../Components/Controls';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Controls', () => {
	let wrapper;
	let mockEvent;
	let mockHandleClick;

	beforeEach(() => {
		wrapper = shallow(<Controls handleClick={mockHandleClick} />);
		mockEvent = { preventDefault: jest.fn() };
		mockHandleClick = jest.fn();
	});

	it('should mock snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('should call handleClick', async () => {
		await wrapper.instance().handleClick(mockEvent);
		expect(mockHandleClick).toHaveBeenCalled();
	});
});
