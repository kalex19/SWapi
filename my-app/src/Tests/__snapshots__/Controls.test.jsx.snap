// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls should mock snapshot 1`] = `ShallowWrapper {}`;
